--- compose-assistant.js	Mon Oct 12 15:00:03 2009
+++ compose-assistant.new	Mon Nov 16 01:39:56 2009
@@ -215,6 +215,11 @@
 						});
 						return;
 					}
+					//If SIPE remove sip: from the username as it breaks the messaging service (jar file)
+					if ((recipient.serviceName == "sipe") && (recipient.value.indexOf("sip:") > -1))
+					{
+						recipient.value = recipient.value.replace("sip:","");
+					}
 					imRecipients.push(recipient);
 				} else {
 					phoneRecipients.push(recipient);
