--- prefsAccountSummary-scene.html	Tue Jan 26 15:16:16 2010
+++ prefsAccountSummary-scene.new	Mon Mar  1 00:57:19 2010
@@ -16,6 +16,12 @@
 					<div id="notificationToggle" x-mojo-element="ToggleButton"></div>
 					<div class="title capitalize">
 						Mostra notifiche
+					</div> 
+				</div>      
+				<div class="palm-row-wrapper">                    
+					<div id="BadCertToggle" x-mojo-element="ToggleButton"></div>
+					<div class="title capitalize" x-mojo-loc=''>
+						Accept Bad Cert
 					</div>                                                               
 				</div> 
 			</div>                
