--- messaging-luna-service.js	Fri Aug 28 12:51:09 2009
+++ messaging-luna-service.js.new	Fri Oct 30 02:25:13 2009
@@ -8,6 +8,7 @@
   displayIdentifier: 'palm://com.palm.display/control',
   telephonyIdentifier: 'palm://com.palm.telephony',
   systemManagerIdentifier: 'palm://com.palm.systemmanager',
+  MessagingPlugins: 'palm://im.libpurpleext.greg',
 
   launchMessaging: function(sceneController,params) {  
 	return sceneController.serviceRequest("palm://com.palm.applicationManager", {
@@ -993,5 +994,34 @@
 		  method: 'setSMSCAddressAndEmailGateway',
 		  parameters: {smscAddr:address, emailGateway: gateway}
 	  });
+	},
+
+  /************************
+   * Methods for LibpurpleAdapter
+   ************************/
+	setserver: function(sceneController,serviceName,ServerName,ServerPort,TLS,ServerLogin) {
+	    return sceneController.serviceRequest(MessagingMojoService.MessagingPlugins, {
+	      method: 'setserver',
+	      parameters: {serviceName:serviceName, ServerName:ServerName, ServerPort:ServerPort, TLS:TLS, ServerLogin:ServerLogin}
+		});  
+	},
+	getserver: function(onSuccess, sceneController, serviceName) {
+		return sceneController.serviceRequest(MessagingMojoService.MessagingPlugins,{
+			method: 'getserver',
+			parameters: {serviceName:serviceName},
+			onSuccess: onSuccess
+		});   	
+	},  
+	loadserver: function(sceneController,serviceName) {
+	    return sceneController.serviceRequest(MessagingMojoService.MessagingPlugins, {
+	      method: 'loadserver',
+	      parameters: {serviceName:serviceName}
+		});  
+	},
+	clearserver: function(sceneController,serviceName) {
+	    return sceneController.serviceRequest(MessagingMojoService.MessagingPlugins, {
+	      method: 'clearserver',
+	      parameters: {serviceName:serviceName}
+		});  
 	}
 };
\ No newline at end of file
