--- global-widget-addressing.css	Mon Jan 18 01:46:14 2010
+++ global-widget-addressing.new	Mon Jan 18 01:46:06 2010
@@ -253,6 +253,46 @@
 	margin: 0px 4px 0px 16px;
 }
 
+.addressing-widget .icon.xfire {                    
+  background: url(../images/accounts/xfire-32x32.png) center left no-repeat;
+}
+
+.addressing-widget .icon.qqim {                    
+  background: url(../images/accounts/qq-32x32.png) center left no-repeat;
+}
+
+.addressing-widget .icon.gwim {                    
+  background: url(../images/accounts/novell-32x32.png) center left no-repeat;
+}
+
+.addressing-widget .icon.irc {                    
+  background: url(../images/accounts/irc-32x32.png) center left no-repeat;
+}
+
+.addressing-widget .icon.lcs {                    
+  background: url(../images/accounts/sametime-32x32.png) center left no-repeat;
+}
+
+.addressing-widget .icon.live {                    
+  background: url(../images/accounts/messenger-32x32.png) center left no-repeat;
+}
+
+.addressing-widget .icon.icq {                    
+  background: url(../images/accounts/icq-32x32.png) center left no-repeat;
+}
+
+.addressing-widget .icon.facebook {                    
+  background: url(../images/accounts/facebook-32x32.png) center left no-repeat;
+}
+
+.addressing-widget .icon.jabber {                    
+  background: url(../images/accounts/jabber-32x32.png) center left no-repeat;
+}
+
+.addressing-widget .icon.sipe {                    
+  background: url(../images/accounts/sipe-32x32.png) center left no-repeat;
+}
+
 .addressing-widget .icon.gmail {                                 
   background: url(../images/accounts/googletalk-32x32.png) center left no-repeat;
 }
