--- prefsSetupAccount-assistant.js	Mon Nov  2 11:14:47 2009
+++ prefsSetupAccount-assistant.js.new	Tue Jan 12 17:07:59 2010
@@ -16,8 +16,76 @@
 		this.handlePasswordKeyUp = this.handlePasswordKeyUp.bindAsEventListener(this);
 		this.removeAccount = this.removeAccount.bindAsEventListener(this);
 		this.confirmRemoveAccount = this.confirmRemoveAccount.bindAsEventListener(this);
+
+		this.tattr = {
+  			trueLabel:  'true' ,
+ 			falseLabel:  'false'
+  		}
+		this.tModel = {
+			value : false,
+ 			disabled: false
+		}
+		this.pattr = {
+  			trueLabel:  'true' ,
+ 			falseLabel:  'false'
+  		}
+		this.pModel = {
+			value : false,
+ 			disabled: false
+		}
+
+		this.params.server = '';
+	       this.params.tls = ''; 
+		this.params.proxy = '';  
     },
 
+    getServerSettings: function(data){
+	
+      this.params.server = data.ServerName;
+      this.params.port = data.ServerPort;
+      this.params.tls = data.ServerTLS;  
+
+      //Update Fields
+      this.serverModel.original = data.ServerName;
+      this.controller.modelChanged(this.serverModel);
+
+      this.portModel.original = data.ServerPort;
+      this.controller.modelChanged(this.portModel);
+
+      if (data.ServerTLS == "false")
+	{
+	      this.tModel.value = false;
+	      this.controller.modelChanged(this.tModel);
+	}
+      else
+	{
+	      this.tModel.value = true;
+	      this.controller.modelChanged(this.tModel);
+	}
+
+	if (this.Messaging.transports.getTransportByName(this.params.domain).className == "sipe"){
+		//Get Server Value
+		this.params.login = data.ServerLogin;
+		this.serverLoginModel.original = data.ServerLogin;
+      		this.controller.modelChanged(this.serverLoginModel);
+
+		//Get Proxy Value
+	      this.params.proxy = data.ServerProxy; 
+
+	      if (data.ServerProxy == "false")
+		{
+		      this.pModel.value = false;
+		      this.controller.modelChanged(this.pModel);
+		}
+	      else
+		{
+		      this.pModel.value = true;
+		      this.controller.modelChanged(this.pModel);
+		}
+	}
+
+    },
+
     setup: function(){
         this.accountId;
         this.createAccountRequest;
@@ -64,9 +132,54 @@
 				this.imLinksElement = this.controller.get('im-links');
 				this.imLinksElement.observe(Mojo.Event.tap, this.handleExtraContentTap);
 			}
+			// Special case for Server Defined Plugins - add extra content below password field
+			if ((this.Messaging.transports.getTransportByName(this.params.domain).className == "sipe") || (this.Messaging.transports.getTransportByName(this.params.domain).className == "jabber")  || (this.Messaging.transports.getTransportByName(this.params.domain).className == "sametime")  || (this.Messaging.transports.getTransportByName(this.params.domain).className == "gwim")) {
+				var ServerDetails = Mojo.View.render({object:{}, template:"prefsSetupAccount/ServerDetails"});				
+				this.controller.get('ServerDetails').update(ServerDetails);
+
+				//Add extra login and proxy field for Office Communicator
+				if (this.Messaging.transports.getTransportByName(this.params.domain).className == "sipe"){
+					//Login
+					var LoginDetails = Mojo.View.render({object:{}, template:"prefsSetupAccount/OfficeCommunicator"});				
+					this.controller.get('LoginDetails').update(LoginDetails);
+
+					//Proxy
+					this.controller.setupWidget('ProxyToggle', this.pattr,this.pModel );
+					this.ProxyToggleChanged = this.ProxyToggleChanged.bindAsEventListener(this);
+					Mojo.Event.listen(this.controller.get('ProxyToggle'),Mojo.Event.propertyChange,this.ProxyToggleChanged);
+				}
+
+				this.controller.setupWidget('TLSToggle', this.tattr,this.tModel );
+				this.TLSToggleChanged = this.TLSToggleChanged.bindAsEventListener(this);
+				Mojo.Event.listen(this.controller.get('TLSToggle'),Mojo.Event.propertyChange,this.TLSToggleChanged);
+
+				//Load Server Details
+				MessagingMojoService.getserver(this.getServerSettings.bind(this),this.controller,this.Messaging.transports.getTransportByName(this.params.domain).className);
+			}
 		} else {
 			this.controller.get('prefHeaderText').update($L('Add an account'));
 			this.saveAccountFn = this.saveAccount.bind(this);
+
+			// Special case for Server Defined Plugins - add extra content below password field
+			if ((this.Messaging.transports.getTransportByName(this.params.domain).className == "sipe") || (this.Messaging.transports.getTransportByName(this.params.domain).className == "jabber") || (this.Messaging.transports.getTransportByName(this.params.domain).className == "sametime") || (this.Messaging.transports.getTransportByName(this.params.domain).className == "gwim")) {
+				var ServerDetails = Mojo.View.render({object:{}, template:"prefsSetupAccount/ServerDetails"});				
+				this.controller.get('ServerDetails').update(ServerDetails);
+
+				//Add extra login field for Office Communicator
+				if (this.Messaging.transports.getTransportByName(this.params.domain).className == "sipe"){
+					//Server
+					var LoginDetails = Mojo.View.render({object:{}, template:"prefsSetupAccount/OfficeCommunicator"});				
+					this.controller.get('LoginDetails').update(LoginDetails);
+
+					//Proxy
+					this.controller.setupWidget('ProxyToggle', this.pattr,this.pModel );
+					this.ProxyToggleChanged = this.ProxyToggleChanged.bindAsEventListener(this);
+					Mojo.Event.listen(this.controller.get('ProxyToggle'),Mojo.Event.propertyChange,this.ProxyToggleChanged);
+				}
+
+				this.controller.setupWidget('TLSToggle', this.tattr,this.tModel );
+				this.TLSToggleChanged = this.TLSToggleChanged.bindAsEventListener(this);
+				Mojo.Event.listen(this.controller.get('TLSToggle'),Mojo.Event.propertyChange,this.TLSToggleChanged);			}
 		}
 		
         var usernameAttributes = {
@@ -106,6 +219,49 @@
 		}
         this.controller.setupWidget('password', passwordAttributes, this.passwordModel);		
 
+        var serverAttributes = {
+            textFieldName: "server",
+            hintText: '',
+            modelProperty: 'original',
+            multiline: false,
+            focus: false,
+			textReplacement: false
+        };
+        this.serverModel = {
+            original: this.params.server || '',
+			disabled: false
+        };
+        this.controller.setupWidget('server', serverAttributes, this.serverModel);
+
+	if (this.Messaging.transports.getTransportByName(this.params.domain).className == "sipe"){
+		var serverLoginAttributes = {
+	            textFieldName: "login",
+	            hintText: '',
+	            modelProperty: 'original',
+	            multiline: false,
+	            focus: false,
+				textReplacement: false
+	        };
+	        this.serverLoginModel = {
+	            original: this.params.login || '',
+				disabled: false
+	        };
+	        this.controller.setupWidget('login', serverLoginAttributes, this.serverLoginModel);
+	 }
+        var portAttributes = {
+            textFieldName: "port",
+            hintText: '',
+            modelProperty: 'original',
+            multiline: false,
+            focus: false,
+			textReplacement: false
+        };
+        this.portModel = {
+            original: this.params.port || '',
+			disabled: false
+        };
+        this.controller.setupWidget('port', portAttributes, this.portModel);
+
 		this.saveButton = this.controller.get('saveAccountButton');
 		var saveButtonAttributes = {
 			disabledProperty: 'disabled',
@@ -129,6 +285,25 @@
 		}
 	},
 	
+    TLSToggleChanged: function(event) {
+	//	this.controller.showAlertDialog({
+	//	    onChoose: function(value) {},
+	//	    title:"The toggle value changed",
+	//	    message:"Toggle value is now: " + event.value,
+	//	    choices:[ {label:'OK', value:'OK', type:'color'} ]
+	//    });
+	},
+
+
+    ProxyToggleChanged: function(event) {
+	//	this.controller.showAlertDialog({
+	//	    onChoose: function(value) {},
+	//	    title:"The toggle value changed",
+	//	    message:"Toggle value is now: " + event.value,
+	//	    choices:[ {label:'OK', value:'OK', type:'color'} ]
+	//    });
+	},
+
 	handlePasswordKeyUp: function(event) {
 		// If enter is pressed then simulate tapping on "Sign In"
 		if (event && Mojo.Char.isEnterKey(event.keyCode)) {
@@ -144,6 +319,12 @@
 				this.passwordTapHandler = undefined;
 			}
 		}
+		// Special case for Server Defined Plugins
+		if ((this.Messaging.transports.getTransportByName(this.params.domain).className == "sipe") || (this.Messaging.transports.getTransportByName(this.params.domain).className == "jabber") || (this.Messaging.transports.getTransportByName(this.params.domain).className == "sametime") || (this.Messaging.transports.getTransportByName(this.params.domain).className == "gwim"))
+		{
+			this.passwordElement.stopObserving(Mojo.Event.tap, this.passwordTapHandler);
+			this.passwordTapHandler = undefined;
+		}
 	},
 	
 	cleanup: function() {
@@ -184,25 +365,51 @@
 	  	statusMessageElement.update($L('Please enter a username'));
 		statusMessageWrapper.show();
 		hasEmptyField = true;
-	  } else if(!accountDetails.password) {
-	  	statusMessageElement.update($L('Please enter a password'));
-		statusMessageWrapper.show();
-		hasEmptyField = true;
-	  } else {
-	  	statusMessageElement.update('');
-		statusMessageWrapper.hide();
-	  }
-	  
+	  } else if((!accountDetails.password) && (this.Messaging.transports.getTransportByName(this.params.domain).className != "irc")) {
+        statusMessageElement.update($L('Please enter a password'));
+        statusMessageWrapper.show();
+        hasEmptyField = true;
+	  } else if((this.Messaging.transports.getTransportByName(this.params.domain).className == "sipe") || (this.Messaging.transports.getTransportByName(this.params.domain).className == "jabber") || (this.Messaging.transports.getTransportByName(this.params.domain).className == "sametime") || (this.Messaging.transports.getTransportByName(this.params.domain).className == "gwim")) {
+		    if(this.Messaging.transports.getTransportByName(this.params.domain).className != "sipe"){
+				if(!accountDetails.port) {
+					statusMessageElement.update($L('Please enter a port'));
+					statusMessageWrapper.show();
+					hasEmptyField = true;
+				}
+			    if(!accountDetails.server) {
+		  	        statusMessageElement.update($L('Please enter a server'));
+			        statusMessageWrapper.show();
+			        hasEmptyField = true;
+			    }
+		    }
+	  }	  
 	  if(hasEmptyField) {
 	  	this.stopLoginSpinner();
 		return;
 	  }
+	  else {
+	  	statusMessageElement.update('');
+		statusMessageWrapper.hide();
+	  }
 	  
       accountDetails.accountTypeId = this.params.accountTypeId;
       accountDetails.domain = this.params.domain;
 
 	  this.startLoginSpinner();
       this.createAccountRequest = MessagingMojoService.createIMAccount(this.controller, accountDetails,this.accountSaved.bind(this),this.createLoginFailed.bind(this));
+
+      //Add new server, port and TLS details
+      if((this.Messaging.transports.getTransportByName(this.params.domain).className == "sipe") || (this.Messaging.transports.getTransportByName(this.params.domain).className == "jabber") || (this.Messaging.transports.getTransportByName(this.params.domain).className == "sametime") || (this.Messaging.transports.getTransportByName(this.params.domain).className == "gwim")) {
+            //setserver: function(sceneController,serviceName,ServerName,ServerPort,TLS)
+
+	     if (this.Messaging.transports.getTransportByName(this.params.domain).className == "sipe"){
+	            MessagingMojoService.setserver(this.controller, this.Messaging.transports.getTransportByName(this.params.domain).className,accountDetails.server,accountDetails.port,this.tModel.value,accountDetails.login,this.pModel.value);
+	     }
+	     else
+	     {
+		     MessagingMojoService.setserver(this.controller, this.Messaging.transports.getTransportByName(this.params.domain).className,accountDetails.server,accountDetails.port,this.tModel.value,'');
+	     }
+      }
     },
 	
 	accountSaved : function(result){
@@ -250,11 +457,21 @@
 	 * UPDATE ACCOUNT METHODS
 	 */
     updateAccount : function(){
-	  if (this.passwordModel.original === PrefsSetupAccountAssistant.kDummyPassword) {
-	  	Mojo.Log.info("updateAccount doing noop b/c password hasn't changed");
-		this.popScene();
-	  	return;
+	  if((this.Messaging.transports.getTransportByName(this.params.domain).className == "sipe") || (this.Messaging.transports.getTransportByName(this.params.domain).className == "jabber") || (this.Messaging.transports.getTransportByName(this.params.domain).className == "sametime") || (this.Messaging.transports.getTransportByName(this.params.domain).className == "gwim")) {
+		  if (this.passwordModel.original === PrefsSetupAccountAssistant.kDummyPassword && this.serverModel.original === this.params.server && this.portModel.original === this.params.port && this.tModel.value === this.params.tls && this.pModel.value === this.params.proxy) {
+		  	Mojo.Log.info("updateAccount doing noop b/c password hasn't changed");
+			this.popScene();
+		  	return;
+		  }
 	  }
+	  else
+	  {
+		  if (this.passwordModel.original === PrefsSetupAccountAssistant.kDummyPassword) {
+		  	Mojo.Log.info("updateAccount doing noop b/c password hasn't changed");
+			this.popScene();
+		  	return;
+		  }
+	  }
 	  var accountDetails = Mojo.View.serializeMojo(this.controller.get('AccountSetupForm'), true);
 	  var statusMessageElement = this.controller.get('StatusMessage');
 	  var statusMessageWrapper = this.controller.get('StatusMessageWrapper');
@@ -267,19 +484,46 @@
 	  	statusMessageElement.update($L('Please enter a password'));
 		statusMessageWrapper.show();
 		hasEmptyField = true;
-	  } else {
-	  	statusMessageElement.update('');
-	    statusMessageWrapper.hide();
+	  } else if((this.Messaging.transports.getTransportByName(this.params.domain).className == "sipe") || (this.Messaging.transports.getTransportByName(this.params.domain).className == "jabber") || (this.Messaging.transports.getTransportByName(this.params.domain).className == "sametime") || (this.Messaging.transports.getTransportByName(this.params.domain).className == "gwim")) {
+		if(this.Messaging.transports.getTransportByName(this.params.domain).className != "sipe"){
+				if(!accountDetails.port) {
+					statusMessageElement.update($L('Please enter a port'));
+					statusMessageWrapper.show();
+					hasEmptyField = true;
+				}
+			    if(!accountDetails.server) {
+		  	        statusMessageElement.update($L('Please enter a server'));
+			        statusMessageWrapper.show();
+			        hasEmptyField = true;
+			    }
+		    }
+
 	  }	  
 	  
 	  if(hasEmptyField) {
 	  	this.stopLoginSpinner();
 		return;
 	  }
+	  else {
+	  	statusMessageElement.update('');
+	    	statusMessageWrapper.hide();
+	  }	
 	  
 	  this.startLoginSpinner();
       this.accountId = this.params.id;
       this.updateAccountRequest = MessagingMojoService.updateAccountPassword(this.controller, this.params.id,accountDetails.password,this.accountUpdated.bind(this),this.updateLoginFailed.bind(this));
+
+      //Update server, port and TLS details
+      if((this.Messaging.transports.getTransportByName(this.params.domain).className == "sipe") || (this.Messaging.transports.getTransportByName(this.params.domain).className == "jabber") || (this.Messaging.transports.getTransportByName(this.params.domain).className == "sametime") || (this.Messaging.transports.getTransportByName(this.params.domain).className == "gwim")) {
+            //setserver: function(sceneController,serviceName,ServerName,ServerPort,TLS)
+	     if (this.Messaging.transports.getTransportByName(this.params.domain).className == "sipe"){
+	            MessagingMojoService.setserver(this.controller, this.Messaging.transports.getTransportByName(this.params.domain).className,accountDetails.server,accountDetails.port,this.tModel.value,accountDetails.login,this.pModel.value);
+	     }
+	     else
+	     {
+		     MessagingMojoService.setserver(this.controller, this.Messaging.transports.getTransportByName(this.params.domain).className,accountDetails.server,accountDetails.port,this.tModel.value,'');
+	     }
+      }
     },
 	
     accountUpdated : function(result) {
@@ -346,7 +590,16 @@
 		this.saveButtonModel.disabled = false;
 		this.controller.modelChanged(this.saveButtonModel);	  
 		this.passwordModel.disabled = false;
-		this.controller.modelChanged(this.passwordModel);	 	
+		this.controller.modelChanged(this.passwordModel);
+		this.serverModel.disabled = false;
+		this.controller.modelChanged(this.serverModel);
+		this.portModel.disabled = false;
+		this.controller.modelChanged(this.portModel); 	
+
+		if (this.Messaging.transports.getTransportByName(this.params.domain).className == "sipe"){
+			this.serverLoginModel.disabled = false;
+			this.controller.modelChanged(this.serverLoginModel);
+		}	 	
 		
 		// the username field is always disabled in edit mode
 		// the delete account button is only visible in edit mode
@@ -364,6 +617,15 @@
 		this.controller.modelChanged(this.saveButtonModel);
 		this.passwordModel.disabled = true;
 		this.controller.modelChanged(this.passwordModel);	 
+		this.serverModel.disabled = true;
+		this.controller.modelChanged(this.serverModel);
+		this.portModel.disabled = true;
+		this.controller.modelChanged(this.portModel); 
+ 
+		if (this.Messaging.transports.getTransportByName(this.params.domain).className == "sipe"){
+			this.serverLoginModel.disabled = true;
+			this.controller.modelChanged(this.serverLoginModel);
+		}
 
 		// the username field is always disabled in edit mode
 		// the delete account button is only visible in edit mode
@@ -398,6 +660,12 @@
 	removeAccount: function() {
 		this.startRemoveSpinner();
         MessagingMojoService.removeAccount(this.controller, this.params.id,this.handleRemoveAccountSuccess.bind(this));        
+
+        //Remove server, port and TLS config
+        if((this.Messaging.transports.getTransportByName(this.params.domain).className == "sipe") || (this.Messaging.transports.getTransportByName(this.params.domain).className == "jabber") || (this.Messaging.transports.getTransportByName(this.params.domain).className == "sametime") || (this.Messaging.transports.getTransportByName(this.params.domain).className == "gwim")) {
+            //clearserver: function(sceneController,serviceName)
+            MessagingMojoService.clearserver(this.controller, this.Messaging.transports.getTransportByName(this.params.domain).className);
+        }
 	},	
 	
 	handleRemoveAccountSuccess: function() {
