--- global-iconography.css	Thu Jul  2 18:06:28 2009
+++ global-iconography.new	Sat May  1 22:50:16 2010
@@ -88,6 +88,60 @@
   float: left;
 }
 
+.palm-account-icon.gadu {
+  background: url(../images/accounts/gadu-32x32.png) center center no-repeat;
+}
+.palm-account-icon.gadu.medium {
+  background: url(../images/accounts/gadu-48x48.png) center center no-repeat;
+}
+.palm-account-icon.xfire {
+  background: url(../images/accounts/xfire-32x32.png) center center no-repeat;
+}
+.palm-account-icon.xfire.medium {
+  background: url(../images/accounts/xfire-48x48.png) center center no-repeat;
+}
+.palm-account-icon.gwim {
+  background: url(../images/accounts/novell-32x32.png) center center no-repeat;
+}
+.palm-account-icon.gwim.medium {
+  background: url(../images/accounts/novell-48x48.png) center center no-repeat;
+}
+.palm-account-icon.irc {
+  background: url(../images/accounts/irc-32x32.png) center center no-repeat;
+}
+.palm-account-icon.irc.medium {
+  background: url(../images/accounts/irc-48x48.png) center center no-repeat;
+}
+.palm-account-icon.lcs {
+  background: url(../images/accounts/sametime-32x32.png) center center no-repeat;
+}
+.palm-account-icon.lcs.medium {
+  background: url(../images/accounts/sametime-48x48.png) center center no-repeat;
+}
+.palm-account-icon.live {
+  background: url(../images/accounts/messenger-32x32.png) center center no-repeat;
+}
+.palm-account-icon.live.medium {
+  background: url(../images/accounts/messenger-48x48.png) center center no-repeat;
+}
+.palm-account-icon.icq {
+  background: url(../images/accounts/icq-32x32.png) center center no-repeat;
+}
+.palm-account-icon.icq.medium {
+  background: url(../images/accounts/icq-48x48.png) center center no-repeat;
+}
+.palm-account-icon.jabber {
+  background: url(../images/accounts/jabber-32x32.png) center center no-repeat;
+}
+.palm-account-icon.jabber.medium {
+  background: url(../images/accounts/jabber-48x48.png) center center no-repeat;
+}
+.palm-account-icon.sipe {
+  background: url(../images/accounts/sipe-32x32.png) center center no-repeat;
+}
+.palm-account-icon.sipe.medium {
+  background: url(../images/accounts/sipe-48x48.png) center center no-repeat;
+}
 .palm-account-icon.aim {
   background: url(../images/accounts/aim-32x32.png) center center no-repeat;
 }
